hamilton/function_modifiers/configuration.py

```diff
@@ -175,6 +175,9 @@ def _get_function_name(self, fn: Callable) -> str:
     def resolve(self, fn, config: Dict[str, Any]) -> Callable:
         if not self.does_resolve(config):
             return None
+        # attaches config keys used to resolve function
+        fn.__config_decorated__ = self._config_used if self._config_used else ["__UNKNOWN__"]
```

Comment:
`is not None` -- I don't think this will be empty but just in case

Reply:
since we don't see a case for `[]` being passed, I think truthiness is fine here -- it's more concise and covers both.